checkfiles/checkfiles.py

```diff
@@ -716,109 +716,112 @@ def check_file(config, session, url, job):
     try:
         file_stat = os.stat(local_path)
     except FileNotFoundError:
-        errors['file_not_found'] = 'File has not been uploaded yet.'
-        if job['run'] < job['upload_expiration']:
-            job['skip'] = True
+        if job['run'] > job['upload_expiration']:
+            errors['file_not_found'] = 'File has not been uploaded yet.'
+        job['skip'] = True
+        return job
+    except OSError:
```

Comment:
when do we get OSError, can we document that here?

Reply:
Added comments